iroha/test_network/src/lib.rs

```diff
@@ -324,6 +326,16 @@ where
         }
     }
 
+    pub async fn stop_async(&mut self) {
+        self.broker
+            .issue_send(iroha_p2p::network::StopSelf(0))
+            .await;
+        if let Some(shutdown) = self.shutdown.take() {
+            shutdown.abort();
+            iroha_logger::error!("Shutting down peer...");
+        }
```

Comment:
Shutting down runtime should stop everything anyway. No need to explicitly stop network.

Reply:
It wont be gentle, but its ok for tests

Reply:
The test `restarted_peer_should_have_the_same_asset_amount()` needs this, as it is shutting down the peer in the middle of the test. Without stopping the network we can't use the same port for another peer.

Reply:
Makes sence. May be we can do it on peer side? (like in iroha crate itself, not in test_network)

Reply:
You can create separate runtime for peer which you want to restart, and just stop its runtime. I kinda dont like this thing (as it requires all tests to do `stop_async` manually, while before it was just drop-checked)

Reply:
No, we don't stop peers in every test manually, just in that one.

Reply:
I know right, but your change kinda requires to stop every peer manually via `stop_async`, while you can just stop runtime for this test (for that peer) and just make `stop` a sync method (and call it via `Drop` trait).

Reply:
That's why I've left `stop()` as it was and made a separate `stop_async()` for one test.

Reply:
Sure, i just say that you can remove `stop_async` and not duplicate things. You can fully restart whole peer, by stopping its runtime in a test.

Reply:
Done.